### PR TITLE
[FIXED JENKINS-34226] No pressed style for OK button on Create Item.

### PR DESCRIPTION
**Pull request by Andrei94**
# Description

See [JENKINS-34226](https://issues.jenkins-ci.org/browse/JENKINS-34226).

The button was using a different style from the one that was used for example in save/apply for in the job config section. I changed it so that it uses the same style as the others.

### Submitter checklist

- [*] JIRA issue is well described
- [*] Link to JIRA ticket in description, if appropriate

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
CC random @jenkinsci/code-reviewers with UI background: @kzantow , @recena, @tfennelly

File: core/src/main/resources/hudson/model/View/newJob.jelly [hunk 1]
```diff
@@ -65,7 +65,7 @@ THE SOFTWARE.
             </j:if>
 
             <div class="footer">
-              <div class="btn-decorator">
+              <div class="yui-button">
```
Comment (recena) [inline core/src/main/resources/hudson/model/View/newJob.jelly]: No more YUI dependency.

File: war/src/main/js/add-item.js [hunk 2]
```diff
@@ -46,7 +46,7 @@ $.when(getItems()).done(function(data) {
 
     function isItemNameEmpty() {
       var itemName = $('input[name="name"]', '#createItem').val();
-      return (itemName === '') ? true : false;
+      return (itemName === '');
```
Comment (recena) [inline war/src/main/js/add-item.js]: What is the relation with the event behavior that you want to change?

Comment (Andrei94) [inline war/src/main/js/add-item.js]: I haven't found one. It was just a small improvement that could be made.

File: core/src/main/resources/hudson/model/View/newJob.jelly (hunk 1 above)
Comment (Andrei94) [inline core/src/main/resources/hudson/model/View/newJob.jelly]: I'm sorry, should I remove the yui dependency?

Comment (recena) [inline core/src/main/resources/hudson/model/View/newJob.jelly]: @Andrei94 I added a new CSS class in order to stop using YUI.

File: war/src/main/js/add-item.js (hunk 2 above)
Comment (recena) [inline war/src/main/js/add-item.js]: Unrelated with the JIRA issue. IMHO, the previous one is more readable.

**Comment by oleg-nenashev** [comment]
@Andrei94 @recena so how do we proceed here?

**Comment by oleg-nenashev** [comment]
@Andrei94 @recena ping

File: core/src/main/resources/hudson/model/View/newJob.jelly (hunk 1 above)
Comment (Andrei94) [inline core/src/main/resources/hudson/model/View/newJob.jelly]: @recena the button is in a `footer` class and hence it doesn't get the YUI style, like the buttons from the other views i.e. the job config step. Here the buttons are part of the table and in the `additem.less` there is reference to the YUI styles.

Comment (Andrei94) [inline core/src/main/resources/hudson/model/View/newJob.jelly]: @recena ping

**Comment by recena** [comment]
My apologies, I lost this PR...

**Comment by oleg-nenashev** [comment]
@recena you have lost it again, I'd guess

**Comment by daniel-beck** [comment]
It is unclear to me why this is labeled _work in progress_, it seems a good code review or two and a test drive away from being merged. Or am I missing something?

**Comment by daniel-beck** [comment]
Okay, I updated the PR branch and tried this. Here are screen shots.

# Before

> OK-old-disabled

> OK-old-enabled

> OK-old-focused

# After (with this PR)

> OK-new-disabled

> OK-new-focused

OK-new-pressed

I would like to note that "old style" buttons with a pressed style exist in job configuration forms, so I'm confused as to the need to restyle these so thoroughly.

> Save-pressed

Not to mention the alignment/margin seems a bit weird as well.

**Comment by daniel-beck** [comment]
> The button was using a different style from the one that was used for example in save/apply for in the job config section. I changed it so that it uses the same style as the others.

Perhaps something changed since this PR was filed (despite there not having been a merge conflict), but it doesn't look like it's just like the others.

**Comment by daniel-beck** [comment]
I noticed a few more problems with the current state:

The button does not stick to the bottom of the screen (which might be deliberate, not sure) when the content is higher than the viewport. Screenshot from before:

> old-fixed

Since there's no area around the button, the grey area below the list is much larger than it would need to be, see previous screen shots.

And finally, there is a pressed, disabled state:

> new-disabled-pressed

**Comment by daniel-beck** [comment]
@Andrei94 I would love to see this PR updated to address review feedback (above), but in the current state it's not something I'd want to merge.

I completely understand if you're no longer interested, and I apologize for our failure to provide timely feedback.

**Review by daniel-beck — CHANGES_REQUESTED** [review]
As above

**Comment by batmat** [comment]
I think this PR will be closed on next pass if we don't hear back in the meantime.

Again, we are **very** sorry we weren't responsive enough initially so this could move forward after it was filed. The reason we are closing this is also because we're trying to fix this mistake by having a better vision on the current PRs and **actually** review them.

Obviously: even after we possibly have closed this, please feel free to just comment here if you'd like to get back to it. We will be happy to reopen anytime.

Thanks for your understanding.

**Comment by batmat** [comment]
Closing already, as latest comment from Daniel saying the same thing was 9 days old. Please refer to my previous anytime if you'd like to see it reopened. Thanks!

**Pull request closed without merging**